ctapipe/io/simteleventsource.py

```diff
@@ -205,12 +206,11 @@ def __generator(self):
                 data.mc.tel[tel_id].time_slice = float(pixel_settings['time_slice'])
 
                 n_pixel = data.r0.tel[tel_id].waveform.shape[-2]
-
-                data.mc.tel[tel_id].photo_electron_image = array_event.get(
-                    'photoelectrons', {}
-                ).get(tel_id)
-                if data.mc.tel[tel_id].photo_electron_image is None:
-                    data.mc.tel[tel_id].photo_electron_image = np.zeros((n_pixel, ), dtype='i2')
+                data.mc.tel[tel_id].photo_electron_image = (
+                    array_event.get('photoelectrons', {})
+                               .get(tel_index, {})
+                               .get('photoelectrons', np.zeros(n_pixel, dtype='float32'))
```

Comment:
is `np.zeros` correct here? or should it be `np.nans` since we have no idea what the values are ... instead of being sure they were zeros.

Reply:
I know it was also `np.zeros` before this PR .. so this PR is in principle not responsible for fixing this .. but still.

Reply:
np.zeros was what was returned in the case of HESSIOEventSource, but I don't have a strong opinion about what to return...

For normal simtel files (prod3+), the np.zeros will never be returned anyway. Only in the case of the prod2 test files